node/packages/aws-lambda-otel-extension/internal/otel-extension-internal-node/index.js

```diff
@@ -198,26 +198,23 @@ const responseHandler = async (span, { res, err }, isTimeout) => {
   if (isTimeout) {
     functionData.error = true;
     functionData.errorCulprit = 'timeout';
-    functionData.errorExceptionType = 'TimeoutError';
-    functionData.errorExceptionMessage = '';
-    functionData.errorExceptionStacktrace =
-      'Function execution exceeded the configured timeout limit.';
+    functionData.errorType = 'TimeoutError';
   } else if (err) {
     functionData.error = true;
     functionData.errorCulprit = err.message;
-    functionData.errorExceptionType = typeof err;
-    functionData.errorExceptionMessage = err.message;
-    functionData.errorExceptionStacktrace = err.stack;
+    functionData.errorType = typeof err;
```

Comment:
I think it doesn't make much sense to use `typeof err` as there's no value in it

I think what we want to know here, is whether it was: _timeout_, _unhandled_, _handled_ or _initialization_ error (and this case we speak of _handled_ case)

Reply:
Okay so in this case we should maybe put unhandled?

Reply:
Got it so really in this file the error should always be marked as handled 👍

Reply:
@medikoo thinking about this a bit more it might actually make sense for this to be marked as `unhandled`. This block of code could be executed in the case that an error is thrown from the lambda function that was not caught by the customers code 🤔

Reply:
> This block of code could be executed in the case that an error is thrown from the lambda function that was not caught by the customers code 🤔

But that'll mean that all errors reported from lambda _unhandled_ while we need to differentiate from those returned from Lambda and those which crash lambda.

So we need type name for the following:
- Timeout
- Lambda crashed during invocation
- Lambda crashed during initialization
- Lambda invocation resolved with an error.

How would name those 4 types? Note here, we're dealing with the last one. And Lambda crashes by AWS are reported as _Unhandled exceptions_

Reply:
🤔 Maybe we should have error types named basically like you outlined above

- `TimeoutError`: Set when we have a timeout error
- `UnhandledInvocation`: Lambda crashed during invocation
- `UnhandledInitialization`: Lambda crashed during initialization
- `UnhandledResolvedError`: Lambda invocation resolved with an error
- `HandledError`: Set if we receive a handled error (Ex. we see a response with a 500 error)

They are kind of verbose but I think they accurately describe what is happening 🤔

Reply:
I think we will have to revisit it at some point.

As ideally if lambdas can be instrumented only with external extension, and over there (not knowing the shape of the response), such case will be reported as _success_.

And for data coherence ideally if internal extension doesn't change the report as produced by the external but enriches instead (so internal could add info on HTTP statuses, and basing on that we show extra alarms to users)